### PR TITLE
Fix a couple of minor issues

### PR DESCRIPTION
**Pull request by remoun**
Fixes some issues I found by running `cargo hack check --each-feature`

* Fix an error when mc-util-telemetry is built without the jaeger feature. Follow-up to #1340.
* Add missing `blocking` feature to `reqwest` dependency.
* Bonus cleanup: Move conditional mods into the associated `cfg_if` block.

### CODE REVIEW

**Review by eranrund — APPROVED** [review]

**Review by wjuan-mob — APPROVED** [review]

File: util/telemetry/src/lib.rs [hunk 1]
```diff
@@ -117,7 +116,9 @@ cfg_if::cfg_if! {
 
             opentelemetry_jaeger::new_pipeline()
                 .with_service_name(service_name)
-                .with_trace_config(Config::default().with_resource(Resource::new(tags)))
+                .with_trace_config(
+                    sdk::trace::Config::default()
+                        .with_resource(sdk::Resource::new(tags)))
```
Comment (garbageslam) [inline util/telemetry/src/lib.rs:121]: Another way to fix this is to move the use statements into the cfg if block with the other use statements for this block

File: attest/net/Cargo.toml [hunk 2]
```diff
@@ -26,7 +26,7 @@ displaydoc = "0.2"
 mbedtls = "0.8.1"
 pem = "1.0"
 percent-encoding = "2.1.0"
-reqwest = { version = "0.10" , default-features = false, features = ["rustls-tls", "gzip"] }
+reqwest = { version = "0.10" , default-features = false, features = ["blocking", "rustls-tls", "gzip"] }
```
Comment (garbageslam) [inline attest/net/Cargo.toml:29]: I think this doesn't do anything unless you also pick the blocking module at the call sites

**Review by garbageslam — CHANGES_REQUESTED** [review]
Make sure the change to reqwest is actually doing something

File: attest/net/Cargo.toml (hunk 2 above)
Comment (remoun) [reply attest/net/Cargo.toml:29]: Right, we use the `reqwest::blocking::Client` without the corresponding feature.

Comment (garbageslam) [reply attest/net/Cargo.toml:29]: oh i see thanks

**Review by garbageslam — APPROVED** [review]

**Pull request merged**